src/Visitors/AddArrayPropertyItem.php

```diff
@@ -19,8 +19,13 @@ public function __construct(
         mixed $value,
     ) {
         list($propertyValue, $propertyType) = $this->getPropertyValue($value);
+
         $this->arrayItem = new ArrayItem($propertyValue);
-        $this->propertyItem = new PropertyProperty($this->name, new Array_([$this->arrayItem]));
+        $arrayNode = new Array_([$this->arrayItem]);
+
+        $this->propertyItem = new PropertyProperty($this->name, $arrayNode);
+        $this->setParentForNewNodeTree($arrayNode, $this->propertyItem);
```

Comment:
let's wrap it into the new parent class method which will call `setParentForNewNodeTree` inside

```suggestion
        $this->propertyItem = $this->prepareNewNode(new PropertyProperty($this->name, $arrayNode));
```